src/conversations/Conversations.ts

```diff
@@ -117,19 +115,33 @@ export default class Conversations {
 
     const decodeMessage = async (
       env: messageApi.Envelope
-    ): Promise<Conversation | Message | null> => {
+    ): Promise<Conversation | DecodedMessage | null> => {
       const contentTopic = env.contentTopic
       if (!contentTopic) {
         return null
       }
 
       if (contentTopic === introTopic) {
-        const msg = await this.client.decodeEnvelope(env)
+        const messageBytes = b64Decode(env.message as unknown as string)
+        const msg = await MessageV1.fromBytes(messageBytes)
         if (!messageHasHeaders(msg)) {
           return null
         }
+        // Decrypt the message to ensure it hasn't been spoofed
+        await msg.decrypt(this.client.legacyKeys)
+        const peerAddress =
+          msg.senderAddress === this.client.address
+            ? msg.recipientAddress
+            : msg.senderAddress
+
+        // Temporarily create a convo to decrypt the message
+        const convo = new ConversationV1(
```

Comment:
I don't love having to create this throwaway conversation instance, but it feels like a necessary consequence of making all decoding tied to the conversation.

Reply:
Yeah, I think this is just something we need to accept. Nice to see that separating Invitation from Message avoids this.